packages/shared/useInViewRef.ts

```diff
@@ -0,0 +1,51 @@
+import { useEffect, useCallback, useState } from 'react';
+import { HTMLElementOrNull, CallbackRef } from './utils/utils';
+
+const config: IntersectionObserverInit = {
+  root: null,
+  rootMargin: '0px 0px 0px 0px',
+  threshold: [0, 1],
+};
+
+/**
+ *
+ * useInViewRef hook
+ *
+ * Returns a mutation observer for a React Ref and true/false when element enters/leaves the viewport. Also fires a callback.
+ *
+ * @param {IntersectionObserverCallback} callback Function that needs to be fired on mutation
+ * @param {IntersectionObserverInit} options
+ */
+function useInViewRef(
+  callback: IntersectionObserverCallback = () => {},
```

Comment:
@maciekgrzybek looks like these comments were never posted, because I didn't click on the "Submit Review" button lol. I cced you a week ago or so haha.

Reply:
Classic, happened to me so many times :D I'm almost sure that there was a reason why I thought it makes more sense to create a new one... Don't remember why now :) I'll check it today tomorrow and fix it if needed :)

Reply:
I might be missing something but how we can pass myRef down to the hook in that case?

Reply:
I think in this case we will have to do some weird state uplifting..? Not sure it abstraction is worth it in this case. But like I said, I might be missing something.